x/poe/bootstrap.go

```diff
@@ -89,14 +89,68 @@ func bootstrapPoEContracts(ctx sdk.Context, k wasmtypes.ContractOpsKeeper, tk tw
 	}
 	ocInitMsg := newOCInitMsg(gs)
 	deposit := sdk.NewCoins(gs.OversightCommitteeContractConfig.EscrowAmount)
-	ocContractAddr, _, err := k.Instantiate(ctx, ocCodeID, systemAdminAddr, systemAdminAddr, mustMarshalJson(ocInitMsg), "oversight_committee", deposit)
+
+	ocMembers := make([]string, len(gs.Engagement))
+	for i, member := range gs.Engagement {
+		ocMembers[i] = member.Address
+	}
+	firstMember, err := sdk.AccAddressFromBech32(ocMembers[0])
+	if err != nil {
+		return sdkerrors.Wrap(err, "first member")
+	}
+
+	ocContractAddr, _, err := k.Instantiate(ctx, ocCodeID, firstMember, systemAdminAddr, mustMarshalJson(ocInitMsg), "oversight_committee", deposit)
 	if err != nil {
 		return sdkerrors.Wrap(err, "instantiate tg trusted circle contract")
 	}
 	poeKeeper.SetPoEContractAddress(ctx, types.PoEContractTypeOversightCommunity, ocContractAddr)
 	if err := k.PinCode(ctx, ocCodeID); err != nil {
 		return sdkerrors.Wrap(err, "pin tg trusted circle contract")
 	}
+
+	tcAdapter := contract.NewTrustedCircleContractAdapter(ocContractAddr, tk, nil)
+	err = tcAdapter.AddVotingMembersProposal(ctx, ocMembers[1:], firstMember)
+	if err != nil {
+		return sdkerrors.Wrap(err, "add voting members proposal")
+	}
+	latest, err := tcAdapter.LatestProposal(ctx)
+	if err != nil {
+		return sdkerrors.Wrap(err, "query latest proposal")
+	}
+	err = tcAdapter.ExecuteProposal(ctx, latest.ID, firstMember)
+	if err != nil {
+		return sdkerrors.Wrap(err, "execute proposal")
+	}
+	// deposit escrow
+	for _, member := range ocMembers[1:] {
+		ocMember, err := sdk.AccAddressFromBech32(member)
+		if err != nil {
+			return sdkerrors.Wrap(err, "oc member")
+		}
+		err = tcAdapter.DepositEscrow(ctx, gs.OversightCommitteeContractConfig.EscrowAmount, ocMember)
+		if err != nil {
+			return sdkerrors.Wrap(err, "deposit escrow")
+		}
+	}
+
+	voters, err := tcAdapter.QueryListVoters(ctx)
+	if err != nil {
+		return sdkerrors.Wrap(err, "query voters")
+	}
+
+	//verify the members list are all voting members
+	notVotingMembers := make(map[string]struct{}, len(ocMembers))
```

Comment:
Looking at this, I am not sure if we need to have a sanity check at all. We can expect the contract to work proper and just verify the member list in an integration test.